include/deal.II/fe/fe_tools.templates.h

```diff
@@ -3243,6 +3243,106 @@ namespace FETools
     return Utilities::invert_permutation(
       hierarchic_to_lexicographic_numbering<dim>(degree));
   }
+
+
+
+  template <int dim>
+  std::pair<std::vector<unsigned int>, std::vector<unsigned int>>
+  cell_to_face_lexicographic(const unsigned int &degree,
+                             const unsigned int &direction,
+                             const bool         &cell_hierarchical_numbering,
+                             const bool         &is_continuous)
+  {
+    AssertThrow(dim > 1, ExcInvalidFEDimension(1, dim));
+    AssertThrow(direction < dim, ExcInvalidFEDimension(direction, dim));
+
+    // n_1d: DoFs per 1D edge (parallel to face)
+    const unsigned int n_1d = degree + 1;
+    // n_ort: DoFs in the direction orthogonal to the face in the combined face
+    // space For discontinuous: 2*n (n for cell0 face, n for cell1 face) For
+    // continuous: 2*n - 1 (shared DoFs)
+    const unsigned int n_ort_1d = 2 * n_1d - (is_continuous ? 1 : 0);
+
+    const unsigned int dofs_per_cell = Utilities::fixed_power<dim>(n_1d);
+    const unsigned int dofs_per_face =
+      Utilities::fixed_power<dim - 1>(n_1d) * n_ort_1d;
```

Comment:
Is this referring to the face? I think this is the total number of DoFs on the patch, as this takes the `dim-1`-dimensional power of `n_1d` and multiplies by the dofs in the orthogonal direction. I would therefore choose another variable name.

Reply:
I will rename it to per_face_patch.